apps/blockchain/lib/blockchain/contract/create_contract.ex

```diff
@@ -119,28 +119,26 @@ defmodule Blockchain.Contract.CreateContract do
       value_in_wei: params.endowment,
       machine_code: params.init_code,
       stack_depth: params.stack_depth,
-      block_interface: BlockInterface.new(params.block_header, state_with_blank_contract.db),
+      block_interface: BlockInterface.new(params.block_header, account_interface.state.db),
       account_interface: account_interface,
       config: params.config
     }
 
     EVM.VM.run(params.available_gas, exec_env)
   end
 
-  @spec init_account(t, EVM.address()) :: EVM.state()
+  @spec init_account(t, EVM.address()) :: AccountInterface.t()
   defp init_account(params, address) do
-    account = Account.get_account(params.account_interface.state, address)
+    {account, _code} = AccountInterface.account(params.account_interface, address)
```

Comment:
Why does `AccountInterface.account` return an `{account, code}` tuple? It seems like it should just return an account, right? As far as I can tell, most places just ignore the `code` part of the tuple.

Reply:
I added a method that returns only account